ksql-streams/src/main/java/io/confluent/ksql/execution/streams/TableGroupByBuilder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import io.confluent.ksql.execution.codegen.CodeGenRunner;
+import io.confluent.ksql.execution.codegen.ExpressionMetadata;
+import io.confluent.ksql.execution.context.QueryContext;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.TableGroupBy;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.serde.KeySerde;
+import java.util.List;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Grouped;
+import org.apache.kafka.streams.kstream.KGroupedTable;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.KeyValueMapper;
+
+public final class TableGroupByBuilder {
+  private TableGroupByBuilder() {
+  }
+
+  public static KGroupedTable<Object, GenericRow> build(
+      final KTable<Object, GenericRow> ktable,
+      final TableGroupBy<?, ?> step,
+      final KsqlQueryBuilder queryBuilder,
+      final GroupedFactory groupedFactory
+  ) {
+    final LogicalSchema sourceSchema = step.getSources().get(0).getProperties().getSchema();
```

Comment:
This pattern of getting the first source is duplicated a lot through the code.  I'm also assuming it would be an error if `getSources` returned none or many sources.  I'm wondering if we can codify this...?

Maybe add a `getSource` method?   This could either be on the `ExecutionStep`, and just default to:

```java
   default ExecutionStep<?> getSource() {
    return Iterables.getOnlyElement(getSources());
  }
```

Or possibly more correctly, add to each step with only a single source. e.g. `TableGroupBy` would have:

```java
   public ExecutionStep<?> getSource() {
    return source;
  }
```

Reply:
I'll go with the latter, since I want a source with all the type info.